client/src/PaginatorContext.js

```diff
@@ -75,24 +123,48 @@ const useJobPaginator = (initialCategory, initialJob) => {
         const message = [];
         if (deleted.length > 0) {
             message.push(deleted.length + " file(s) have been removed from the job since the last edit.\n");
-            setOrder(getNonRemovedPages(order, deleted));
+            dispatch({
+                type: "UPDATE_ORDER",
+                payload: getNonRemovedPages(order, deleted),
+            });
         }
 
         if (added.length > 0) {
             message.push(added.length + " file(s) have been added to the job since the last edit.\n");
-            setOrder(getAddedPages(order, added));
+            dispatch({
+                type: "UPDATE_ORDER",
+                payload: getAddedPages(order, added),
+            });
         }
 
         if (message.length) {
             alert(message.join());
         }
     };
 
-    const loadJob = async () => {
-        const { order } = await fetchJSON(getJobUrl(category, job));
-        setOrder(order);
-        setCurrentPage(0);
-        updatePagesByStatus(await fetchJSON(getJobUrl(category, job, "/status")));
+    const setInitials = (initialCategory, initialJob) => {
+        dispatch({
+            type: "UPDATE_CATEGORY",
+            payload: initialCategory,
+        });
+        dispatch({
+            type: "UPDATE_JOB",
+            payload: initialJob,
+        });
+    };
+
+    const loadJob = async (initialCategory, initialJob) => {
+        setInitials(initialCategory, initialJob);
+        const { order } = await fetchJSON(getJobUrl(initialCategory, initialJob));
```

Comment:
At this moment, error handling is handled at the fetchJSON. But, I do think it needs to be refactored to be consistent on all calls. I remember talking about how the error handling was inconsistent and we probably need a task for the refactor.

Reply:
Yes, agreed, error handling is inconsistent. I think it's reasonable to set this up so that fetchJSON throws an exception when something goes wrong... but then the calling code should trap that exception and do something with it, or else we'll just get unhandled exceptions blowing up the React environment (if I remember correctly how this all works). As you say, though, probably worth treating that as a separate task.